src/phpDocumentor/Descriptor/Interfaces/InheritsFromElement.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Descriptor\Interfaces;
+
+use phpDocumentor\Reflection\Fqsen;
+
+interface InheritsFromElement
+{
+    /**
+     * Returns the element from which this element inherits, or null if it doesn't inherit any information.
+     *
+     * @return ElementInterface|string|Fqsen|null
```

Comment:
Why do we accept string here?

Reply:
Should be investigated in a subsequent PR, I have copied this from existing code and dared not change it

Reply:
AFAIC, we should investigate how we can reduce this to a single type with, for example, proxies or reference objects. But: subsequent PR